lms/static/scripts/frontend_apps/utils/test/api-test.js

```diff
@@ -788,4 +793,12 @@ describe('usePolledAPIFetch', () => {
     assert.isFalse(isLoading());
     assert.isFalse(isRefreshing());
   });
+
+  it('cancels polling after max refreshes are reached', () => {
+    mockFetchFinished();
+
+    const wrapper = createComponent({ maxRefreshes: 0 });
+
+    assert.equal(wrapper.text(), 'Canceled');
+  });
```

Comment:
We need some kind of test that exercises the behavior of the internal `refreshing` flag. eg. Start a poll with a refresh limit, change the path/params mid-poll, verify that the total number of refreshes is allowed to exceed the limit (since it was reset at one point).